### PR TITLE
AMIGAOS: Static builds preferred

### PR DESCRIPTION
**Pull request by raziel-**
Reasons:
- Shared objects aren't really shared on AmigaOS.
(Once two programs load the same .so, it will be available twice in memory, instead of sharing the already available one)
- To make the program run for everyone i need to provide a subdir (SObjs/) which holds all the .so's used while compiling (which are not available in a clean OS install), otherwise most people have to go hunting for the correct .so's online.
Even worse is the fact that, if users have older (incompatible) .so's installed, they might experience crashes.
- There is no benefit in building ScummVM with shared objects (at least on AmigaOS), because even *if* a new lib version (i.e. libPNG) would be made available as an .so (and features new stuff), it won't be mandatory to immediately switch to it, because ScummVM probably won't take advantage of the new features as fast anyway.
So, a rebuilt is always better (imho).

In short:
Switchting to static builds to reduce user grief, crash reports, a little less work to create the package and as a bonus have ScummVM start a little faster.

### CODE REVIEW

**Comment by bluegr** [comment]
Sounds logical. Since you're the maintainer for the AmigaOS port, you know better, @raziel-. I assume this will work when cross-compiling (i.e. with buildbot builds)?

**Comment by lordhoto** [comment]
Thanks @raziel-.

**Comment by raziel-** [comment]
Thank you @lordhoto
